tests/MulterTests.js

```diff
@@ -0,0 +1,392 @@
+    testMdPath: fluid.module.resolvePath("%kettle/tests/data/multer/test.md")
+};
+
+fluid.defaults("kettle.tests.multer.handler.single", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        "multerSingle": {
+            middleware: "{server}.infusionMulterSingle"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "kettle.tests.multerHandlerSingle"
+        }
+    }
+});
+
+// Sends info about the uploaded file
+kettle.tests.multerHandlerSingle = function (request) {
+    request.events.onSuccess.fire(request.req.file);
+};
+
+fluid.defaults("kettle.tests.multer.handler.array", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        "multerArray": {
+            middleware: "{server}.infusionMulterArray"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "kettle.tests.multerHandlerArray"
+        }
+    }
+});
+
+// Sends info about the uploaded array of files
+kettle.tests.multerHandlerArray = function (request) {
+    request.events.onSuccess.fire(request.req.files);
+};
+
+fluid.defaults("kettle.tests.multer.handler.fields", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        "multerFields": {
+            middleware: "{server}.infusionMulterFields"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "kettle.tests.multerHandlerField"
+        }
+    }
+});
+
+// Sends info about both the body and the uploaded files
+kettle.tests.multerHandlerField = function (request) {
+    request.events.onSuccess.fire({body: request.req.body, files: request.req.files});
+};
+
+fluid.defaults("kettle.tests.multer.handler.imageOnly", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        "multerImageOnly": {
+            middleware: "{server}.infusionMulterImageOnly"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "kettle.tests.multerHandlerSingle"
+        }
+    }
+});
+
+fluid.defaults("kettle.tests.multer.handler.diskStorage", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        "multerDiskStorage": {
+            middleware: "{server}.infusionMulterDiskStorage"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "kettle.tests.multerHandlerSingle"
+        }
+    }
+});
+
+kettle.tests.multer.testDefs = [{
+    name: "Multer tests",
+    expect: 29,
+    config: {
+        configName: "kettle.tests.multer.config",
+        configPath: "%kettle/tests/configs"
+    },
+    components: {
+        singleFileUpload: {
+            type: "kettle.test.request.formData",
+            options: {
+                path: "/multer-single",
+                method: "POST",
+                formData: {
+                    files: {
+                        "file": kettle.tests.multer.testFilePaths.testTxtPath
```

Comment:
I ended up putting them onto a holder component on the testDef structure, since they need to be reused multiple times, and referring to them via IoC references. Acceptable?